main_test.go

```diff
@@ -14,43 +14,67 @@ import (
 
 func TestRun(t *testing.T) {
 	t.Run("allow unmatched", func(t *testing.T) {
+		tmpDir := t.TempDir()
 		cfg := &config.Config{AllowUnmatched: true}
 
 		output, err := runWithConfig(t, cfg)
 		require.NoError(t, err)
 		assert.Contains(t, output, "No issues found.")
+
+		// Same config with --sarif-report-path flag
+		output, err = runWithConfig(t, cfg, tmpDir+"/no_issues.sarif")
+		require.NoError(t, err)
+		file, err := os.ReadFile("internal/sops/testdata/sarif_test_outputs/no_issues.sarif")
+		require.NoError(t, err)
+		assert.Equal(t, output, string(file))
```

Comment:
Let's separate out the sarif tests into dedicated test cases instead of running twice with different arguments. Right now it's not obvious if a failing test is caused by issues specific to sarif.

Reply:
made them as  dedicated tests